open-craft/edx-platform
# Discussion XBlock - initial version

Initial version - PR for code review.

Testing instructions:
*0. Do not install this changeset yet.*
1. Create a course.
2. Add a couple of units and a couple of discussions (stock modules) to it.
3. Post some posts, comments and responses and otherwise interact with discussions: follow/unfollop, upvote/downvote, etc.
4. Install this changeset.
5. Make sure LMS reruns installing dependencies; if not, at least run `pip install -r requeirements/edx/local.txt` - it'll install discussion XBlock and remove discussion XModule.
6. Make sure all the threads, comments and responses are still visible, upvote/downvote count is correct, etc.
7. Check that still can post threads/comments/responses, upvote/downvote, etc.

Note that tests are failing in this branch so far - there's OC-1624 to address that.
